extensions/positron-r/src/r-installation.ts

```diff
@@ -25,20 +25,82 @@ export interface RMetadataExtra {
 	 * https://github.com/posit-dev/positron/issues/2659
 	 */
 	readonly current: boolean;
+
+	/**
+	 * How did we discover this R binary?
+	 */
+	readonly reasonDiscovered: ReasonDiscovered[] | null;
+}
+
+/**
+ * Enum represents how we discovered an R binary.
+ */
+export enum ReasonDiscovered {
+	affiliated,
+	registry,
+	/* eslint-disable @typescript-eslint/naming-convention */
+	PATH,
+	HQ,
+	/* eslint-enable @typescript-eslint/naming-convention */
+	adHoc,
+	user
+}
+
+/**
+ * Enum represents why we rejected an R binary.
+ */
+export enum ReasonRejected {
+	invalid = 'invalid',
+	unsupported = 'unsupported',
+	nonOrthogonal = 'non-orthogonal'
+}
```

Comment:
While expanding the treatment of "reasons", I also did what @jmcphers requested here re: enum handling: https://github.com/posit-dev/positron/pull/4878/files#r1785400981